src/components/ArmyBuildPane.jsx

```diff
@@ -17,30 +17,42 @@ const Container = styled.div`
 `;
 
 const ArmyBuildPane = () => {
+
   const [unitList, setUnitList] = useState({});
   const [listIndex, setListIndex] = useState(1);
 
   const addUnit = unit => {
+    unit.selected_upgrades = {};
     setUnitList({
       ...unitList,
       [listIndex]: unit
```

Comment:
To not mutate incoming argument:
```suggestion
      [listIndex]: {
          ...unit,
          selected_upgrades: {}
      }
```